library/objective-c/EnvoyHTTPStreamImpl.m

```diff
@@ -84,11 +84,21 @@ static envoy_headers toNativeHeaders(EnvoyHeaders *headers) {
 
 #pragma mark - C callbacks
 
+// Return whether a callback should be allowed to continue with execution.
```

Comment:
Do you mind adding a description here about the new logic. i.e this is done in order to ensure that only one 'terminal' callback is fired.